.github/workflows/build-docsite.yaml

```diff
@@ -0,0 +1,34 @@
+name: Build the Ansible docsite
+
+on:
+  push:
+    branches:
+      - 'main'
+  workflow_dispatch:
+
+jobs:
+  build-docsite:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out the repository
+      uses: actions/checkout@v3
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: "3.10"
+    - name: Install requirements
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install -r requirements.in -c requirements.txt
+    - name: Generate assets for the docs landing page
+      run: python build.py
+    - name: Copy sitemaps, redirects, and robots.txt files
+      run: |
+        cp *-sitemap.xml output
+        cp .htaccess output
+        cp robots.txt output
+    - name: Deploy to the prod branch
+      uses: JamesIves/github-pages-deploy-action@v4
+      with:
+        folder: output
+        branch: prod
```

Comment:
I was thinking about this one but kept the `JamesIves/github-pages-deploy-action` action in place because it didn't seem to have the need for additional permissions.

And we aren't really deploying anything via github pages. This simply copies the generated HTML onto the `prod` branch so that the RH internal Jenkins job can deploy the contents from there to the web server. The Jenkins job clones the `prod` branch and then rsyncs everything across.

What if we abandon the `prod` branch altogether and simply build a tarball with the build action? We can update the Jenkins job to fetch the tarball, unpack it, and then rsync the generated HTML and other assets to the web server.

Reply:
I've been overthinking it. There's no need for the `prod` branch or a tarball that we'd use curl to fetch via the GH API. We can simply build the docsite as part of that Jenkins job.

I removed the deploy action altogether. We can produce an archive of the generated HTML and assets in case anyone wants to look.

Reply:
I suppose that's fine. My preference would be to push the built artifact to the remote target directly, though.